### DISCUSSION TITLE
Create `@directus/shared` for extensions

### DISCUSSION THREAD

**Discussion by rijkvanzanten** [directus/directus#5820] [Feature Requests]
We should have a single shared package that can be used to share parts of code that are reused between app/api/packages and extensions. This can include types, utility functions, vue components, constants

**Comment by n0script22** [comment]
finding myself here and came to conclusion that there is no point to create extensions at this state. really disappointing

**Reply by rijkvanzanten** [reply]
`@directus/shared` already exists, and `@directus/extensions-sdk` re-exports the types/util fn/vue components/constants etc relevant for extensions. 

> came to conclusion that there is no point to create extensions at this state. really disappointing

Care to elaborate on the _why_? Lets keep the discussions to constructive feedback 👍🏻

**Reply by n0script22** [reply]
_Sorry that i was a bit harsh on that._

I integrated stripe and paypal endpoints and i had to impersionate a "SystemUser" to fire specific db-queries to create order items.
I had to copy tons of methods to get the correct permissions for my systemuser accountability object, because i was not able to inject **getPermissions** (api/src/utils/get-permissions.ts) and its missing in the shared packages.

**Reply by danielband** [reply]
It should be possible to [bypass](https://github.com/directus/directus/discussions/8182) permissions by not passing the accountability object to the ItemService. That said, I have not tried it yet.

`const service = new ItemsService('collection', { schema });`

But I agree, there are a few utility function that are not exposed like `getAccountabilityForRole`.

**Reply by rijkvanzanten** [reply]
@danielband is correct 🙂 

The main reason most of those internal methods aren't exposed yet is that they tend to change fairly often. Every we fix or tweak something in the internals that touch that method, it's a new breaking change for extensions. Adding methods to shared needs to be done very very carefully, as any change in those exposed methods can cause huge headaches down the line
